### PR TITLE
Run kyuubi on kubernetes - make basic dockerfile and example

### PR DESCRIPTION
**Pull request by yaooqinn**
### _Why are the changes needed?_

```
# Usage:
# Start:
#   kubectl apply -f https://github.com/NetEase/kyuubi/blob/master/docker/example.yml
# Connect:
#   kubectl port-forward kyuubi-example --address localhost 10009:10009
# Or
#   kubectl exec -it kyuubi-example -- /bin/bash
#
#   ${SPARK_HOME}/bin/beeline -u 'jdbc:hive2://localhost:10009'
# Logging
#   kubectl logs -f kyuubi-example
```

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

kubectl apply -f https://raw.githubusercontent.com/yaooqinn/kyuubi/docker/docker/example.yaml

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @pan3793

File: docker/example.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Usage:
+# Start:
+#   kubectl apply -f https://raw.githubusercontent.com/NetEase/kyuubi/master/docker/example.yaml
```
Comment (pan3793) [inline docker/example.yaml:20]: NetEase/kyuubi => apache/incubator-kyuubi

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    useradd -u ${kyuubi_uid} -g root kyuubi && \
+    mkdir -p ${KYUUBI_HOME} ${KYUUBI_LOG_DIR} ${KYUUBI_PID_DIR} ${KYUUBI_WORK_DIR_ROOT} && \
+    chmod ug+rw -R ${KYUUBI_HOME} && \
+    chmod a+rwx -R ${KYUUBI_WORK_DIR_ROOT} && \
+    rm -rf /var/cache/apt/*
+
+COPY bin ${KYUUBI_HOME}/bin
+COPY conf ${KYUUBI_HOME}/conf
+COPY jars ${KYUUBI_HOME}/jars
+COPY externals/engines/spark ${KYUUBI_HOME}/externals/engines/spark
```
Comment (pan3793) [inline docker/Dockerfile:53]: COPY dist/bin  dist/conf ... ?

Comment (yaooqinn) [reply docker/Dockerfile:53]: you shall cd `bin`

Comment (pan3793) [reply docker/Dockerfile:53]: Let's make it consistent with the usage
> `docker build -f docker/Dockerfile -t yaooqinn/kyuubi:tagname .`

**Review by pan3793 — APPROVED** [review]
LGTM

**Pull request merged**